code/GameEngine/Networking/Rpc.cs

```diff
@@ -90,4 +143,94 @@ private static void InvokeRpc( in ObjectMessageMsg message, in TypeDescription t
 
 		Caller = oldCaller;
 	}
+
+	/// <summary>
+	/// Try to find a method name string from the supplied index.
+	/// </summary>
+	/// <param name="index"></param>
+	internal static string FindMethodName( int index )
+	{
+		if ( _indexToMethodName.TryGetValue( index, out var methodName ) )
+		{
+			return methodName;
+		}
+
+		var methods = GetAllMethodNames();
+
+		try
+		{
+			methodName = methods.ElementAt( index );
+			_indexToMethodName[index] = methodName;
+			_methodNameToIndex[methodName] = index;
+			return methodName;
+		}
+		catch
+		{
+			throw new( $"Unknown Static RPC method with index '{index}'" );
+		}
+	}
+
+	/// <summary>
+	/// Try to find an index from the supplied method name.
+	/// </summary>
+	/// <param name="methodName"></param>
+	internal static int FindMethodIndex( string methodName )
+	{
+		if ( _methodNameToIndex.TryGetValue( methodName, out var index ) )
+		{
+			return index;
+		}
+
+		var methods = GetAllMethodNames();
+		var success = false;
+		index = 0;
+			
+		foreach ( var m in methods )
+		{
+			if ( m == methodName )
+			{
+				_methodNameToIndex[methodName] = index;
+				_indexToMethodName[index] = methodName;
+				success = true;
+				break;
+			}
+
+			index++;
+		}
+
+		if ( !success )
+		{
+			throw new( $"Unindexed RPC method '{methodName}'" );
+		}
+
+		return index;
+	}
+
+	static Dictionary<string, int> _methodNameToIndex = new();
+	static Dictionary<int, string> _indexToMethodName = new();
+
+	[Event.Hotload]
+	static void OnHotload()
+	{
+		_methodNameToIndex.Clear();
+		_indexToMethodName.Clear();
+	}
+
+	static IEnumerable<string> GetAllMethodNames()
+	{
+		var staticMethods = TypeLibrary
+			.GetMethodsWithAttribute<BroadcastAttribute>()
+			.Select( ( m ) => $"{m.Method.TypeDescription.FullName}.{m.Method.Name}" );
+		
+		var instanceMethods = TypeLibrary.GetTypes()
+			.SelectMany( x => x.Members )
+			.OfType<MethodDescription>()
+			.Where( x => !x.IsStatic && ( x.Attributes.OfType<BroadcastAttribute>().Any() || x.Attributes.OfType<AuthorityAttribute>().Any() ) )
+			.Select( ( m ) => m.Name );
+
+		return staticMethods
+			.Concat( instanceMethods )
+			.Distinct()
+			.Order();
```

Comment:
This method shouldn't exist. We should be able to get an Identifier from MethodDescription, and look up straight from TypeLibrary. We shouldn't need to duplicate this stuff when it's already all stored in TypeLibrary.

Reply:
Neither `MemberDescription` or `MethodDescription` have an identifier - or are ya saying we should add it in?

Reply:
I think my latest commit should be good then - got rid of that in favor of using a `FastHash()` of the fully qualified method name 👍

Reply:
TypeLibrary already has this built in via MemberDescription.Ident - we shouldn't be re-inventing the wheel here.